src/Stratis.Bitcoin.Features.PoA/PoAConsensusOptions.cs

```diff
@@ -25,15 +26,15 @@ public class PoAConsensusOptions : ConsensusOptions
 
         /// <summary>Makes federation members kick idle members.</summary>
         /// <remarks>Requires voting to be enabled to be set <c>true</c>.</remarks>
-        public bool AutoKickIdleMembers { get; set; }
+        public bool AutoKickIdleMembers { get; protected set; }
```

Comment:
Hmmm whats the reasoning to make all of these `protected`? Now we have to pass more and more things into the constructor? I understand the base class perhaps but I think this is necessary?

Reply:
If we keep them `public` it would be possible to modify the properties after constructing the class - i.e. it would still be mutable.

> Now we have to pass more and more things into the constructor? 

Doing so seems to actually not be that bad in practice. The code looks slightly neater.